### ISSUE TITLE
Windows Loading ELF file as Raw binary - autodetection of format fails with exception

### ISSUE THREAD

**Issue by tbennett6421** [NationalSecurityAgency/ghidra#3036]
**Describe the bug**
On windows, attempts to load an elf file do not populate the proper format

**To Reproduce**
On a Windows 10 system,

1. File > Import File:
2. Choose an elf file
3. Format is listed as Raw Binary

An error is listed inside the project window

**Expected behavior**
The format should be selected as ELF

**Screenshots**
![image](https://user-images.githubusercontent.com/6843296/118347018-9093c780-b505-11eb-9d5d-390680e3b7c7.png)

**Attachments**
If applicable, please attach any files that caused problems or log files generated by the software.

**Environment (please complete the following information):**
 - OS:
 ```
systeminfo /fo csv | ConvertFrom-Csv | select OS*, System*, Hotfix* | Format-List

OS Name             : Microsoft Windows 10 Pro
OS Version          : 10.0.19042 N/A Build 19042
```
 - Java Version:
```
java --version
java 16.0.1 2021-04-20
Java(TM) SE Runtime Environment (build 16.0.1+9-24)
Java HotSpot(TM) 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)
```
 - Ghidra Version:
 `9.2.3`
 - Ghidra Origin: 
 `https://ghidra-sre.org/ghidra_9.2.3_PUBLIC_20210325.zip`

**Additional context**
These same binaries load just fine in `*nix` based OS's including Kali Linux and Ubuntu 20.04

**Comment by tbennett6421** [comment]
Contents of user log
```
2021-05-14	22:48:23	INFO	(LoggingInitialization) Using log config file: jar:file:/C:/Users/root/Downloads/ghidra/Ghidra/Framework/Generic/lib/Generic.jar!/generic.log4j.xml
2021-05-14	22:48:23	INFO	(LoggingInitialization) Using log file: C:\Users\root\.ghidra\.ghidra_9.2.3_PUBLIC\application.log
2021-05-14	22:48:23	INFO	(Preferences) Loading user preferences: C:\Users\root\.ghidra\.ghidra_9.2.3_PUBLIC\preferences
2021-05-14	22:48:25	INFO	(ClassSearcher) Class search complete (1407 ms)
2021-05-14	22:48:25	INFO	(SSLContextInitializer) Initializing SSL Context
2021-05-14	22:48:25	INFO	(SecureRandomFactory) Initializing Random Number Generator...
2021-05-14	22:48:25	INFO	(SecureRandomFactory) Random Number Generator initialization complete: SHA1PRNG
2021-05-14	22:48:25	INFO	(ApplicationTrustManagerFactory) Trust manager disabled, cacerts have not been set
2021-05-14	22:48:25	INFO	(GhidraRun) User root started Ghidra.
2021-05-14	22:48:26	DEBUG	(RecoverySnapshotMgrPlugin) Recovery snapshot timer set to 5 minute(s)
2021-05-14	22:48:26	INFO	(DefaultProject) Opening project: C:\Users\root\ghidra_cases\hackaday-u
2021-05-14	22:49:45	ERROR	(LoaderService) Unexpected Loader exception from Executable and Linking Format (ELF) java.lang.RuntimeException: java.lang.ExceptionInInitializerError
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:61)
			at ghidra.app.util.bin.format.elf.ElfHeader.createElfHeader(ElfHeader.java:104)
			at ghidra.app.util.opinion.ElfLoader.findSupportedLoadSpecs(ElfLoader.java:109)
			at ghidra.app.util.opinion.LoaderService.getSupportedLoadSpecs(LoaderService.java:47)
			at ghidra.app.util.opinion.LoaderService.getAllSupportedLoadSpecs(LoaderService.java:71)
			at ghidra.plugin.importer.ImporterUtilities.importSingleFile(ImporterUtilities.java:312)
			at ghidra.plugin.importer.ImporterUtilities.showImportDialog(ImporterUtilities.java:187)
			at ghidra.plugin.importer.ImporterUtilities.lambda$showImportDialog$0(ImporterUtilities.java:138)
			at ghidra.util.task.TaskLauncher$2.run(TaskLauncher.java:117)
			at ghidra.util.task.Task.monitoredRun(Task.java:124)
			at ghidra.util.task.TaskRunner.lambda$startTaskThread$0(TaskRunner.java:104)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
			at java.base/java.lang.Thread.run(Thread.java:831)
			Caused by: java.lang.ExceptionInInitializerError
			at net.sf.cglib.core.KeyFactory$Generator.generateClass(KeyFactory.java:166)
			at net.sf.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
			at net.sf.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:216)
			at net.sf.cglib.core.KeyFactory$Generator.create(KeyFactory.java:144)
			at net.sf.cglib.core.KeyFactory.create(KeyFactory.java:116)
			at net.sf.cglib.core.KeyFactory.create(KeyFactory.java:108)
			at net.sf.cglib.core.KeyFactory.create(KeyFactory.java:104)
			at net.sf.cglib.proxy.Enhancer.<clinit>(Enhancer.java:69)
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:45)
			... 13 more
			Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @32bc66dd
			at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
			at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
			at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
			at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
			at net.sf.cglib.core.ReflectUtils$2.run(ReflectUtils.java:56)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
			at net.sf.cglib.core.ReflectUtils.<clinit>(ReflectUtils.java:46)
			... 22 more
			
2021-05-14	22:49:45	ERROR	(LoaderService) Unexpected Loader exception from New Executable (NE) java.lang.RuntimeException: java.lang.NoClassDefFoundError: Could not initialize class net.sf.cglib.proxy.Enhancer
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:61)
			at ghidra.app.util.bin.format.mz.DOSHeader.createDOSHeader(DOSHeader.java:100)
			at ghidra.app.util.bin.format.ne.NewExecutable.<init>(NewExecutable.java:46)
			at ghidra.app.util.opinion.NeLoader.findSupportedLoadSpecs(NeLoader.java:70)
			at ghidra.app.util.opinion.LoaderService.getSupportedLoadSpecs(LoaderService.java:47)
			at ghidra.app.util.opinion.LoaderService.getAllSupportedLoadSpecs(LoaderService.java:71)
			at ghidra.plugin.importer.ImporterUtilities.importSingleFile(ImporterUtilities.java:312)
			at ghidra.plugin.importer.ImporterUtilities.showImportDialog(ImporterUtilities.java:187)
			at ghidra.plugin.importer.ImporterUtilities.lambda$showImportDialog$0(ImporterUtilities.java:138)
			at ghidra.util.task.TaskLauncher$2.run(TaskLauncher.java:117)
			at ghidra.util.task.Task.monitoredRun(Task.java:124)
			at ghidra.util.task.TaskRunner.lambda$startTaskThread$0(TaskRunner.java:104)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
			at java.base/java.lang.Thread.run(Thread.java:831)
			Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.sf.cglib.proxy.Enhancer
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:45)
			... 14 more
			
2021-05-14	22:49:45	ERROR	(LoaderService) Unexpected Loader exception from Portable Executable (PE) java.lang.RuntimeException: java.lang.NoClassDefFoundError: Could not initialize class net.sf.cglib.proxy.Enhancer
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:61)
			at ghidra.app.util.bin.format.pe.PortableExecutable.createPortableExecutable(PortableExecutable.java:85)
			at ghidra.app.util.opinion.PeLoader.findSupportedLoadSpecs(PeLoader.java:79)
			at ghidra.app.util.opinion.LoaderService.getSupportedLoadSpecs(LoaderService.java:47)
			at ghidra.app.util.opinion.LoaderService.getAllSupportedLoadSpecs(LoaderService.java:71)
			at ghidra.plugin.importer.ImporterUtilities.importSingleFile(ImporterUtilities.java:312)
			at ghidra.plugin.importer.ImporterUtilities.showImportDialog(ImporterUtilities.java:187)
			at ghidra.plugin.importer.ImporterUtilities.lambda$showImportDialog$0(ImporterUtilities.java:138)
			at ghidra.util.task.TaskLauncher$2.run(TaskLauncher.java:117)
			at ghidra.util.task.Task.monitoredRun(Task.java:124)
			at ghidra.util.task.TaskRunner.lambda$startTaskThread$0(TaskRunner.java:104)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
			at java.base/java.lang.Thread.run(Thread.java:831)
			Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.sf.cglib.proxy.Enhancer
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:45)
			... 13 more
			
2021-05-14	22:49:45	ERROR	(LoaderService) Unexpected Loader exception from Old-style DOS Executable (MZ) java.lang.RuntimeException: java.lang.NoClassDefFoundError: Could not initialize class net.sf.cglib.proxy.Enhancer
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:61)
			at ghidra.app.util.bin.format.mz.DOSHeader.createDOSHeader(DOSHeader.java:100)
			at ghidra.app.util.bin.format.mz.OldStyleExecutable.<init>(OldStyleExecutable.java:41)
			at ghidra.app.util.opinion.MzLoader.findSupportedLoadSpecs(MzLoader.java:72)
			at ghidra.app.util.opinion.LoaderService.getSupportedLoadSpecs(LoaderService.java:47)
			at ghidra.app.util.opinion.LoaderService.getAllSupportedLoadSpecs(LoaderService.java:71)
			at ghidra.plugin.importer.ImporterUtilities.importSingleFile(ImporterUtilities.java:312)
			at ghidra.plugin.importer.ImporterUtilities.showImportDialog(ImporterUtilities.java:187)
			at ghidra.plugin.importer.ImporterUtilities.lambda$showImportDialog$0(ImporterUtilities.java:138)
			at ghidra.util.task.TaskLauncher$2.run(TaskLauncher.java:117)
			at ghidra.util.task.Task.monitoredRun(Task.java:124)
			at ghidra.util.task.TaskRunner.lambda$startTaskThread$0(TaskRunner.java:104)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
			at java.base/java.lang.Thread.run(Thread.java:831)
			Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.sf.cglib.proxy.Enhancer
			at generic.continues.ContinuesFactory.create(ContinuesFactory.java:45)
			... 14 more
```

**Comment by astrelsky** [comment]
Try a jdk >= 11 <= 15. It seems there are some incompatibilities with jdk 16. That exception shouldn't be occurring.

**Comment by tbennett6421** [comment]
This still occurs when using JDK 15

**Comment by ryanmkurtz** [comment]
By Ghidra latest, do you mean the 10.0.3 release or are you running out of the master branch?

**Comment by tbennett6421** [comment]
10.0.3_public. I also just tried jdk14 with the same error

**Comment by ryanmkurtz** [comment]
In `support/launch.properties`, can you uncomment out this line and try again?

```
#VMARGS=-DContinuesInterceptor.disabled=true
```

**Comment by tbennett6421** [comment]
I don't appear to have a line setting an option for the ContinuesIntercepter. The only thing of note is the following commented out block.

# The ContinuesInterceptor allows the import process to proceed if parsing corrupted headers
# generates uncaught exceptions.  Its usage has been deprecated and will be removed in a future
# release of Ghidra.  It is disabled by default.
#VMARGS=-DContinuesInterceptor.enabled=true

**Comment by ryanmkurtz** [comment]
Ah yes, can you uncomment that line and try again?

**Comment by ryanmkurtz** [comment]
We recently disabled that legacy feature by default in 10.0.3 with the hopes that we can fully remove it in a future release of Ghidra, based on user feedback like this.  We'll have to investigate what exact issue it caused in your case.  Can you attach your binary?

**Comment by tbennett6421** [comment]
I used binaries from the following projects.

https://github.com/wrongbaud/hackaday-u
https://github.com/Maijin/radare2-workshop-2015

specifically
https://github.com/wrongbaud/hackaday-u/blob/master/session-one/exercises/c1
https://github.com/Maijin/radare2-workshop-2015/blob/master/IOLI-crackme/bin-win32/crackme0x00.exe

**Comment by tbennett6421** [comment]
[bins.zip](https://github.com/NationalSecurityAgency/ghidra/files/7244358/bins.zip)

**Comment by ryanmkurtz** [comment]
With 10.0.3, I didn't have an issue importing `c1`.

For `crackme0x00.exe`, this is something we have already fixed and will be available in 10.0.4.

**Comment by tbennett6421** [comment]
Awesome, that is great to hear!
